app/services/metrics/merge_time/per_user_project.rb

```diff
@@ -0,0 +1,67 @@
+module Metrics
+  module MergeTime
+    class PerUserProject < BaseService
+      BATCH_SIZE = 500
+
+      def initialize(interval = nil)
+        @interval = interval
+      end
+
+      def call
+        process
+      end
+
+      private
+
+      def process
+        ActiveRecord::Base.transaction do
+          entities = Hash.new(0)
+          filtered_pull_requests.find_each(batch_size: BATCH_SIZE).lazy.each do |pull_request|
+            entity = find_user_project(pull_request.owner, pull_request.project)
+            entities[entity] += 1
+            merge_time = calculate_merge_time(pull_request)
+
+            create_or_update_metric(entity, merge_time)
+          end
+          calculate_avg(entities)
+        end
+      end
+
+      def filtered_pull_requests
+        Events::PullRequest.includes(:project, owner: :users_projects)
+                           .where(opened_at: metric_interval)
+      end
+
+      def calculate_avg(entities)
+        entities.reject { |_entity, count| count == 1 }.each do |entity, count|
+          Metric.find_by!(ownable: entity, created_at: metric_interval).tap do |metric|
```

Comment:
we should use the value_timestamp attr instead created_at

Reply:
And in the `create_or_update_metric` too right?